proposals/gc/MVP.md

```diff
@@ -322,27 +322,54 @@ Tentatively, support a type of guaranteed unboxed scalars.
 * `ref.is_i31` checks whether a reference is an i31
   - `ref.is_i31 : [anyref] -> [i32]`
 
+* `br_on_non_null <labelidx>` branches if a reference is not null
+  - `br_on_non_null $l : [t0* (ref null ht)] -> [t0*]`
```

Comment:
for symmetry (with `br_on_null`, and the others here) should this return the input perhaps?
```suggestion
  - `br_on_non_null $l : [t0* (ref null ht)] -> [t0* (ref null ht)]`
```

Reply:
Well, br_on_null does not pass on the null reference to the label, so the symmetry is that br_on_non_null inversely does not return it. We could have designed br_on_null differently, but it's too late for that.

Is there a use case where you'd actually use the null?

Reply:
I have no specific use case in mind. Yeah, I suppose it's not useful. And good point about `br_on_null` not passing the value.